android/beagle/src/main/java/br/com/zup/beagle/android/context/operations/exception/ExceptionFactory.kt

```diff
@@ -0,0 +1,57 @@
+package br.com.zup.beagle.android.context.operations.exception
+
+import br.com.zup.beagle.android.context.operations.exception.strategy.ExceptionOperationTypes
+import br.com.zup.beagle.android.context.operations.exception.strategy.ExceptionParameterTypes
+import br.com.zup.beagle.android.context.operations.exception.strategy.ExceptionTypes
+import br.com.zup.beagle.android.context.operations.operation.Operation
+import br.com.zup.beagle.android.context.operations.strategy.string.StringOperationTypes
+
```

Comment:
What do you think to put all constants here?

```suggestion
private const val NOT_FOUND_MESSAGE = "Could not resolve your operation. Did you check if the Operation is valid?:: $details"
```

Reply:
This is a Prototype of exceptions, this needs to be improved as Test exceptions, when we found a common solution the Strings will be moved to a constant file

Reply:
Done, move all messages to ENUM Type, there is formatt in the messages, so is better use in enum than string.format